### PR TITLE
Update revenue data when change occur related to donation

### LINKED ISSUE #5297: Revenue should be updated when donation status changes
## Details
We inserted the [`give_revenue`](https://github.com/impress-org/givewp/issues/5257) table in the core which will use to generate revenue reports which will help to improve performance. Revenue should update in database when donation status changes.

## Acceptance Criteria
-   [x] Edit revenue in database when donation status changes.

### PR DESCRIPTION
**Pull request by ravinderk**
Resolves #5297 

## Description
Record revenue in give_reveneue table when inserting a new donation and delete it automatically when deleting donation.

## Affects
- Added cascade delete rule to `give_revenue` table

## Pre-review Checklist
-   [x] Acceptance criteria satisfied and marked in a related issue
-   [x] Relevant `@since` tags included in DocBlocks
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

## Testing Instructions
Test this pr with https://github.com/impress-org/give-funds/pull/31

Complete following step before testing:
- [x] Delete database table `give_revenue`, `give_fund`, `give_fund_form_relationship`
- [x] Delete `give_database_migrations` from option table
- [x] Open any WordPressadmin page

### CODE REVIEW

File: src/Revenue/Repositories/Revenue.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+ *
+ * Use this class to get data from "give_revenue" table.
+ *
+ * @since 2.9.0
+ */
+class Revenue {
+	/**
+	 * Revenue constructor
+	 */
+	public function constructor() {
```
Comment (alaca) [inline src/Revenue/Repositories/Revenue.php:18]: Remove the empty constructor if you don't plan to use it.

File: src/Revenue/Repositories/Revenue.php [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	}
+
+	/**
+	 * Validate revenue data for insertion.
+	 *
+	 * @since 2.9.0
+	 *
+	 * @param array $array
+	 */
+	private function validateDataFormInsertion( $array ) {
```
Comment (alaca) [inline src/Revenue/Repositories/Revenue.php:59]: Typo in the method name?

**Review by alaca — CHANGES_REQUESTED** [review]
There is a couple of minor issues, but overall, looks good! 
Good work!  :+1:

File: src/Revenue/Repositories/Revenue.php (hunk 1 above)
Comment (ravinderk) [reply src/Revenue/Repositories/Revenue.php:18]: fixed: https://github.com/impress-org/givewp/compare/f2d051c89917...bdedb19969be

File: src/Revenue/Repositories/Revenue.php (hunk 2 above)
Comment (ravinderk) [reply src/Revenue/Repositories/Revenue.php]: fixed: https://github.com/impress-org/givewp/compare/e05f93d70f3c...b140fdebe02d

**Pull request merged**